src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -170,6 +171,34 @@ public static Note parseNote(String noteString) throws ParseException {
             throw new ParseException(Note.MESSAGE_CONSTRAINTS);
         }
 
-        return new Note(noteString);
+        return new Note(trimmedNoteString);
+    }
+
+    /**
+     * Parses a {@code String suggestionType} into a {@code SuggestionType}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code suggestionType} is invalid.
+     */
+    public static SuggestionType parseSuggestionType(String suggestionType) throws ParseException {
+        requireNonNull(suggestionType);
+        String trimmedSuggestionType = suggestionType.trim();
+        if (!SuggestionType.isValidSuggestionType(trimmedSuggestionType)) {
+            throw new ParseException(SuggestionType.MESSAGE_CONSTRAINTS);
+        }
+        return new SuggestionType(trimmedSuggestionType);
+    }
+
+
+    /**
+     * Parses {@code Collection<String> tags} into a {@code Set<Tag>}.
```

Comment:
Possible typo. Should `Collection<String> tags` be `Collection<String> suggestionTypes` instead? Similar for the `Set<Tag>`.